src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -147,6 +147,9 @@ public async Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document docu
             //
             // this design's weakness is that each side don't have enough information to narrow down works to do. it will most likely always do more works than needed.
             // sometimes way more than it is needed. (compilation)
+
+            // group diagnostics by their diagnostics span
+            // invariant: later code gathers & runs CodeFixProviders for diagnostics with one identical diagnostics span (that get's later set as CodeFixCollection's TextSpan)
```

Comment:
minor grammatical suggestion :)
```suggestion
            // invariant: later code gathers & runs CodeFixProviders for diagnostics with one identical diagnostics span (that gets set later as CodeFixCollection's TextSpan)
```